src/toast/io/observation_hdf_save.py

```diff
@@ -0,0 +1,753 @@
+        times (str):  The name of the shared timestamp field.
+        force_serial (bool):  If True, do not use HDF5 parallel support,
+            even if it is available.
+
+    Returns:
+        (str):  The full path of the file that was written.
+
+    """
+    log = Logger.get()
+    env = Environment.get()
+    parallel = have_hdf5_parallel()
+    if force_serial:
+        parallel = False
+
+    if obs.name is None:
+        raise RuntimeError("Cannot save observations that have no name")
+
+    timer = Timer()
+    timer.start()
+    log_prefix = f"HDF5 save {obs.name}: "
+
+    namestr = f"{obs.name}_{obs.uid}"
+    hfpath = os.path.join(dir, f"{namestr}.h5")
+    hfpath_temp = f"{hfpath}.tmp"
+
+    # Create the file and get the root group
+    hf = None
+    hgroup = None
+    vtimer = Timer()
+    vtimer.start()
+    if parallel:
+        hf = h5py.File(hfpath_temp, "w", driver="mpio", comm=obs.comm.comm_group)
+        hgroup = hf
+        log.verbose_rank(
+            f"{log_prefix}  Opened temp file {hfpath_temp} in parallel in",
+            comm=obs.comm.comm_group,
+            timer=vtimer,
+        )
+    elif obs.comm.group_rank == 0:
+        hf = h5py.File(hfpath_temp, "w")
+        hgroup = hf
+        log.verbose_rank(
+            f"{log_prefix}  Opened temp file {hfpath_temp} on rank zero in",
+            comm=None,
+            timer=vtimer,
+        )
+
+    save_comm = None
+    if parallel:
+        save_comm = obs.comm.comm_group
+
+    shared_group = None
+    detdata_group = None
+    intervals_group = None
+    if parallel or obs.comm.group_rank == 0:
+        # Record the software versions and config
+        hgroup.attrs["toast_version"] = env.version()
+        if config is not None:
+            hgroup.attrs["job_config"] = json.dumps(config)
+        hgroup.attrs["toast_format_version"] = 0
+
+        # Observation properties
+        hgroup.attrs["observation_name"] = obs.name
+        hgroup.attrs["observation_uid"] = obs.uid
+
+        obs_all_dets = json.dumps(obs.all_detectors)
+        obs_all_det_sets = "NONE"
+        if obs.all_detector_sets is not None:
+            obs_all_det_sets = json.dumps(obs.all_detector_sets)
+        obs_all_samp_sets = "NONE"
+        if obs.all_sample_sets is not None:
+            obs_all_samp_sets = json.dumps(
+                [[str(x) for x in y] for y in obs.all_sample_sets]
+            )
+        hgroup.attrs["observation_detectors"] = obs_all_dets
+        hgroup.attrs["observation_detector_sets"] = obs_all_det_sets
+        hgroup.attrs["observation_samples"] = obs.n_all_samples
+        hgroup.attrs["observation_sample_sets"] = obs_all_samp_sets
+
+        log.verbose_rank(
+            f"{log_prefix}  Wrote observation attributes (parallel={parallel}) in",
+            comm=save_comm,
+            timer=vtimer,
+        )
+
+        # Instrument properties
+        inst_group = hgroup.create_group("instrument")
```

Comment:
Why don't the telescope and site information get dumped with the rest of the metadata?

Reply:
The idea in separating these is that the instrument information is currently stored separately in the `Observation` class (it is not part of the internal metadata dictionary).  And this instrument information is needed to construct the observation.  Keeping them separate allows us to instantiate the observation when loading the data prior to importing the metadata.